src/app/core/entity/model/entity.ts

```diff
@@ -302,7 +301,7 @@ export class Entity {
    */
   public copy(generateNewId: boolean = false): this {
     const other = new (this.getConstructor())(this._id);
-    Object.assign(other, cloneDeep(this));
+    Object.assign(other, this);
```

Comment:
There is a problem with that approach.
When we do a deep copy, we might make a "too deep" copy: E.g. the configurable enum values cannot be matched anymore because in the basic autocomplete component we use object equality but due to the deep copy this does not work anymore.
So I suggest that instead the subclasses take care of creating a proper copy that is as deep as it needs to be (this is anyway what we are already doing).

Reply:
Sounds slightly risky to rely on the sub-classes to remember this, but I understand the point.

Isn't this impossible to cover config-defined array properties then?

Reply:
Good point but then we have a slightly bigger topic to cover, I moved this to #1805